test/api/exception_test.dart

```diff
@@ -25,4 +28,40 @@ void main() {
   });
 
   // NetworkException.toString: see "API network errors" test in core_test.dart
+
+  test('MalformedServerResponseException shows JSON parsing location', () {
+    const jsonResponse = '''
+      {
+        "result": "success",
+        "queue_id": "1234567890",
+        "last_event_id": -1,
+        "properties": {
+          "realm_uri": "https://example.zulipchat.com"
+        }
+      }
+    ''';
+
+    try {
+      Map<String, String> wrongType =
+        (jsonDecode(jsonResponse) as Map<String, dynamic>)
+          .map((key, value) => MapEntry(key, value.toString()));
+
+      wrongType['properties']! as Map<String, dynamic>;
+      fail('Should have thrown');
+    } catch (e, stackTrace) {
+      final exception = MalformedServerResponseException(
```

Comment:
A more end-to-end test will be more valuable. `MalformedServerResponseException` already has a bunch of tests in `test/api/core_test.dart`.

Reply:
We should then leave a comment similar to this one above:

```
// NetworkException.toString: see "API network errors" test in core_test.dart
```

to indicate that the tests live elsewhere.